app/src/main/java/de/mud/terminal/VDUBuffer.java

```diff
@@ -359,8 +359,8 @@ public synchronized void insertLine(int l, int n, boolean scrollDown) {
       if (n > (bottom - top)) n = (bottom - top);
       int size = bottom - l - (n - 1);
       if(size < 0) size = 0;
-      cbuf = new char[size][];
-      abuf = new long[size][];
+      cbuf = new char[size][width];
```

Comment:
i was debating between doing this and L391 (for `cbuf`) vs. adding something like

```java
for (int i = 0; i < cbuf.length; i++) {
    if (cbuf[i] != null) {
        continue;
    }
    cbuf[i] = new char[width];
    Arrays.fill(cbuf[i], ' ');
}
```

just before [L477](https://github.com/connectbot/connectbot/blob/c4f21c10af8b0abde12ccfc3f20060362fe9e124/app/src/main/java/de/mud/terminal/VDUBuffer.java#L477). `abuf` is fine since it will be zero-initialized. Let me know what you prefer.